src/server/v2.0/handler/handler.go

```diff
@@ -0,0 +1,81 @@
+// Copyright Project Harbor Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package handler
+
+import (
+	"encoding/json"
+	"log"
+	"net/http"
+	"net/http/httptest"
+	"strings"
+
+	"github.com/go-openapi/errors"
+	"github.com/goharbor/harbor/src/server/middleware"
+	"github.com/goharbor/harbor/src/server/v2.0/restapi"
+	"github.com/justinas/alice"
+)
+
+// New returns http handler for API V2.0
+func New() http.Handler {
+	h, api, err := restapi.HandlerAPI(restapi.Config{
+		InnerMiddleware: innerMiddleware,
+		ArtifactAPI:     NewArtifactAPI(),
+	})
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	api.ServeError = serveError
+
+	return h
+}
+
+func innerMiddleware(next http.Handler) http.Handler {
+	return alice.New(middleware.RequestID()).Then(next)
+}
+
+type apiError struct {
+	Code    int32  `json:"code"`
+	Message string `json:"message"`
+}
+
+type ociError struct {
+	Code    string `json:"code"`
+	Message string `json:"message"`
+}
+
+// transform openapi error response to OCI error response format
```

Comment:
What's the consideration of doing such a transformation? If it is for the purpose of letting the client identify and render the error, why the management API should return such errors? If I'm not mistaken, so far, swagger API only covers Harbor mgmt API.

Reply:
Before executing operation handler, go-swagger will bind a parameters object to a request and validate the request, it will return directly when bind and validate failed.

The response format of the default ServeError implementation does not match the OCI error response format which used by our operation handlers.

So we needed to convert the format to the OCI error response format.

Reply:
Changed to return our `common error` rather than OCI error.